Add option to not activate connection immediately

Currently when we run
https://github.com/jcronenberg/agama/blob/813a755848a75de1478374d0c3dc49558b08e38f/rust/agama-migrate-wicked/src/migrate.rs#L38
the following line is always executed
https://github.com/jcronenberg/agama/blob/813a755848a75de1478374d0c3dc49558b08e38f/rust/agama-dbus-server/src/network/nm/client.rs#L102
But because the migration is supposed to often be run from a container where the interfaces aren't available an error
`[ERROR] Could not add/update the connection en0: D-Bus service error`
gets logged even though everything else went well. (See e.g. our integration testsuite)

To avoid this I see 2 possibilities: 
* add a flag inside the network model if the connection should be activated or not
* modify the `write()` and `add_or_update_connection()` functions to take a parameter if connections should be activated or not

--- a/rust/migrate-wicked/src/interface.rs
+++ b/rust/migrate-wicked/src/interface.rs
@@ -1,3 +1,4 @@
+use crate::MIGRATION_SETTINGS;
 use agama_dbus_server::network::model::{
     self, IpConfig, IpRoute, Ipv4Method, Ipv6Method, MacAddress,
 };
@@ -113,8 +114,15 @@ impl Interface {
             id: self.name.clone(),
             interface: Some(self.name.clone()),
             ip_config: ip_config.ip_config,
+            status: model::Status::Down,
             ..Default::default()
         };
+
+        if let Some(settings) = MIGRATION_SETTINGS.get() {
+            if settings.activate_connections {
+                base.status = model::Status::Up;
+            }
+        }
 
         if let Some(dummy) = &self.dummy {
             if let Some(address) = &dummy.address {

--- a/rust/migrate-wicked/src/main.rs
+++ b/rust/migrate-wicked/src/main.rs
@@ -49,6 +49,10 @@ pub enum Commands {
         /// Run migration without sending connections to NetworkManager (can be run without NetworkManager installed)
         #[arg(long, global = true)]
         dry_run: bool,
+
+        /// Activate connections immediately
+        #[arg(long, global = true)]
+        activate_connections: bool,
     },
 }
 
@@ -69,6 +73,7 @@ async fn run_command(cli: Cli) -> anyhow::Result<()> {
                 .set(MigrationSettings {
                     continue_migration: true,
                     dry_run: false,
+                    activate_connections: true,
                 })
                 .expect("MIGRATION_SETTINGS was set too early");
 
@@ -89,11 +94,13 @@ async fn run_command(cli: Cli) -> anyhow::Result<()> {
             paths,
             continue_migration,
             dry_run,
+            activate_connections,
         } => {
             MIGRATION_SETTINGS
                 .set(MigrationSettings {
                     continue_migration,
                     dry_run,
+                    activate_connections,
                 })
                 .expect("MIGRATION_SETTINGS was set too early");
 
@@ -123,6 +130,7 @@ impl Termination for CliResult {
 struct MigrationSettings {
     continue_migration: bool,
     dry_run: bool,
+    activate_connections: bool,
 }
 
 static MIGRATION_SETTINGS: OnceCell<MigrationSettings> = OnceCell::const_new();

